### PR TITLE
ttl: add resource group for ttl query

### LINKED ISSUE #40928: Add resource limitation for TTL queries.

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #40928

Problem Summary:

Add resource group `tidb_ttl` for the ttl query. The user can tune the configuration for this resource group through `alter resource group`.

I was running some experiments to get a more proper RU value. 

### Check List

Tests

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

WIP

### Release note

```release-note
Attach resource group for the TTL queries. The user can modify the priority and limitation of the TTL through configuring the resource group.
```

### CODE REVIEW

File: session/bootstrap.go [hunk 1]
```diff
@@ -537,6 +537,10 @@ const (
     	key(parent_table_id, create_time),
     	key(create_time)
 	);`
+
```
Comment (BornChanger) [inline session/bootstrap.go:540]:
```suggestion
CreateDefaultResourceGroup = "CREATE RESOURCE GROUP IF NOT EXISTS `default` RRU_PER_SEC=0 WRU_PER_SEC=0;"
```

File: session/bootstrap.go [hunk 2]
```diff
@@ -2263,6 +2270,13 @@ func upgradeToVer111(s Session, ver int64) {
 	doReentrantDDL(s, CreateTTLJobHistory)
 }
 
+func upgradeToVer112(s Session, ver int64) {
+	if ver >= version112 {
+		return
+	}
+	doReentrantDDL(s, CreateTTLResourceGroup)
```
Comment (BornChanger) [inline session/bootstrap.go:2277]:
```suggestion
	doReentrantDDL(s, CreateTTLResourceGroup)
	doReentrantDDL(s, CreateDefaultResourceGroup)
```

File: session/bootstrap.go [hunk 3]
```diff
@@ -2373,6 +2387,8 @@ func doDDLWorks(s Session) {
 	mustExecute(s, CreateTTLTask)
 	// Create tidb_ttl_job_history table
 	mustExecute(s, CreateTTLJobHistory)
+	// Create tidb_ttl resource group
+	mustExecute(s, CreateTTLResourceGroup)
```
Comment (BornChanger) [inline session/bootstrap.go:2391]:
```suggestion
	mustExecute(s, CreateTTLResourceGroup)
	mustExecute(s, CreateDefaultResourceGroup)
```

File: session/bootstrap.go [hunk 4]
```diff
@@ -537,6 +537,10 @@ const (
     	key(parent_table_id, create_time),
     	key(create_time)
 	);`
+
+	// CreateTTLResourceGroup creates the resource group for ttl
+	// TODO: set a more proper value for the RU_PER_SECOND
+	CreateTTLResourceGroup = `CREATE RESOURCE GROUP tidb_ttl RRU_PER_SEC=10000000 WRU_PER_SEC=10000000`
```
Comment (BornChanger) [inline session/bootstrap.go:543]:
```suggestion
	CreateTTLResourceGroup = `CREATE RESOURCE GROUP IF NOT EXISTS tidb_ttl RRU_PER_SEC=10000000 WRU_PER_SEC=10000000`
```

Comment (YangKeao) [reply session/bootstrap.go]: Fixed.

File: session/bootstrap.go [hunk 5]
```diff
@@ -548,6 +548,10 @@ const (
     	key(parent_table_id, create_time),
     	key(create_time)
 	);`
+
+	// CreateTTLResourceGroup creates the resource group for ttl
+	// TODO: set a more proper value for the RU_PER_SECOND
+	CreateTTLResourceGroup = `CREATE RESOURCE GROUP IF NOT EXISTS tidb_ttl RU_PER_SEC=10000000`
```
Comment (YangKeao) [inline session/bootstrap.go:554]: I set a super big value for the `tidb_ttl`, so the ttl is not limited by default. If the users want to limit the TTL, they could alter the resource group `tidb_ttl`. Is it a good pattern?

**Comment by glorv** [comment]
+ @bb7133  This seems to be a short term workaround for a specific propose and the implementation is somewhat tricky. Do we really need to merge it into master?

File: session/bootstrap.go (hunk 5 above)
Comment (glorv) [reply session/bootstrap.go:554]: You can user `CREATE RESOURCE GROUP IF NOT EXISTS tidb_ttl RU_PER_SEC=10000 BURSTABLE` instead.

**Pull request closed without merging**